app/src/org/commcare/utils/AndroidArrayDataSource.java

```diff
@@ -0,0 +1,24 @@
+package org.commcare.utils;
+
+import android.content.Context;
+
+import org.commcare.util.ArrayDataSource;
+
+/**
+ * ArrayDataSource that uses the Android 'values' resources
+ */
+
+public class AndroidArrayDataSource implements ArrayDataSource {
+
+    private Context context;
+
+    public AndroidArrayDataSource(Context context){
+        this.context = context;
+    }
+
+    @Override
+    public String[] getArray(String key) {
+        int resourceId = context.getResources().getIdentifier(key, "array", context.getPackageName());
```

Comment:
Hm, I think this should probably fall back somehow in its current form, or throw a much more explicit error if it fails.

Before we had a guaranteed no-error, since the array was fetched from ID resource. Now it could get removed and our error message would be incredibly vague. Especially since this can now fail due to a change in commcare-core that is quite invisible to the end-user here.